catalyst/dl/experiment/config.py

```diff
@@ -345,6 +351,17 @@ def get_loaders(self, stage: str) -> "OrderedDict[str, DataLoader]":
             assert isinstance(overridden_loader_params, dict), \
                 f"{overridden_loader_params} should be Dict"
 
+            sampler_params = samplers_params.pop(name, None)
```

Comment:
we should also check, that user's datasets and samplers have same names,

for example, what would be if we have
datasets: `{"train": ..., "valid": ...}` and samplers `{"train": ..., "valid000": ...}`?

Reply:
btw, the same logic should be implemented with loaders

Reply:
No, you don't. It's already done by default. 

You can see above that we are only iterated by the names of the datasets `for name, ds_ in datasets.items()`. And for each of them, we're trying to get a loader and a sampler

Reply:
please, add a validation check for our Catalyst.todo for this case